docs/docker_run.rst

```diff
@@ -0,0 +1,190 @@
+that should be mount into a container. These class can also be using in conjection
+with other part of the girder_work docker infrastructure, for example providing a location
+where a file should be downloaded to. See `Downloading files from Girder`_. When using
+the ``Volume`` class a list on instances is provided as the value for the ``volumes``
+parameter, girder_work will take care of ensuring that these volumes are mount. In the
+example below we are creating a ``Volume`` instance and passing it as a container
+argument to provide the mounted location to the container. girder_worker will take
+care of transforming the instance into the approriate path inside the container.
+
+.. code-block:: python
+
+    vol = Volume('/home/docker/data', '/mnt/docker/')
+    docker_run.delay('my/image', pull_image=True, volumes=[vol], container_args=[vol])
+
+Temporary Volume
+^^^^^^^^^^^^^^^^
+
+A ``TemporaryVolume`` class is provided representing a temporary directory on the host
+machine that is mounted into the container. ``TemporaryVolume.default`` holds a default
+instance that is used as the default location for many other parts of the girder_worker docker
+infrastructure for example when downloading file. See `Downloading files from Girder`_.
+However, it can also be using explicity for example, here it is being passed as a
+container argument for use within a container. Again, girder_worker will take care
+of transforming the ``TemporaryVolume`` instance into the appropriate path inside
+the container, so the container entrypoint will simply recieve a path.
+
+.. code-block:: python
+
+    vol = Volume('/home/docker/data', '/mnt/docker/')
+    docker_run.delay('my/image', pull_image=True, container_args=[TemporaryVolume.default])
+
+Note that we don't have to add the instance to the ``volumes`` parameter as it is
+automatically added to the list of volumes to mount.
+
+Downloading files from Girder
+-----------------------------
+
+Accessing files held in girder from within a container is straight forward using
+the ``GirderFileIdToVolume`` utility class. One simply provides the file id as
+a argument to the constructor and pass the instance as a container argument.
+
+.. code-block:: python
+
+    docker_run.delay('my/image', pull_image=True,
+        container_args=[GirderFileIdToVolume(file_id)])
+
+The ``GirderFileIdToVolume`` instance will take care of downloading the file from
+Girder and passes the path it was download to the docker containers entrypoint as an argument.
+
+If no ``volume`` parameter is specified then the file will be downloading to the
+task temporary volume. The file can also be downloaded to a specific ``Volume`` by
+specifying a volume parameter, as follows:
+
+.. code-block:: python
+
+    vol = Volume(host_path, container_path)
+    docker_run.delay('my/image', pull_image=True,
+        container_args=[GirderFileIdToVolume(file_id,volume=myvol)])
```

Comment:
Are `vol` and `myvol` meant to be the same here? If not, where is `myvol` coming from?

Reply:
This is what is known as a typo, I will fix it.

Reply:
Fixed